### PR TITLE
feat: MLIBZ-2389 Delta Set implementation

### PR DESCRIPTION
**Pull request by yuliya-guseva**
#### Description
Update Delta Set realisation

#### Changes
Updated Delta Set mechanism for `DataStore#pull`

#### Tests
Instrumented, Manual

### CODE REVIEW

File: java-api-core/src/com/kinvey/java/store/BaseDataStore.java [hunk 1]
```diff
@@ -389,10 +413,22 @@ public void pushBlocking() throws IOException {
...
         Preconditions.checkArgument(client.isInitialize(), "client must be initialized.");
         Preconditions.checkArgument(client.getSyncManager().getCount(getCollectionName()) == 0, "InvalidOperation. You must push all pending sync items before new data is pulled. Call push() on the data store instance to push pending items, or purge() to remove them.");
-
-        KinveyAbstractReadResponse<T> response = new KinveyAbstractReadResponse<T>();
         query = query == null ? client.query() : query;
+        if (deltaSetCachingEnabled && query.getSkip() == 0 && query.getLimit() == 0) {
+            return getBlockingDeltaSync(query);
+        } else {
+            return pullBlockingNoDeltaSync(query);
+        }
```
Comment (vinaygahlawat) [inline java-api-core/src/com/kinvey/java/store/BaseDataStore.java]: Why does one method name start with `get`, and the other with `pull`?

Comment (yuliya-guseva) [reply java-api-core/src/com/kinvey/java/store/BaseDataStore.java]: Updated.

File: java-api-core/src/com/kinvey/java/network/NetworkManager.java [hunk 2]
```diff
@@ -362,6 +375,22 @@ public Get getBlocking(Query query, List<T> cachedItems, boolean deltaSetCaching
...
+    /**
+     * Get blocking with Delta Sync implementation
+     * @param query Query to get
+     * @param lastRequestTime Last request time
+     * @return
+     * @throws IOException
+     */
+    public QueryCacheGet queryCacheGetBlocking(Query query, String lastRequestTime) throws IOException {
+        Preconditions.checkNotNull(query);
+        QueryCacheGet pull = new QueryCacheGet(query, myClass, lastRequestTime);
```
[the remark this answers is not in the document]
Comment (yuliya-guseva) [reply java-api-core/src/com/kinvey/java/network/NetworkManager.java]: Updated.

File: java-api-core/src/com/kinvey/java/network/NetworkManager.java [hunk 3]
```diff
@@ -875,6 +904,52 @@ public Subscribe subscribe(String deviceId) throws IOException {
         }
     }
 
+    public class QueryCacheGet extends AbstractKinveyQueryCacheReadRequest<T> {
```
[the remark this answers is not in the document]
Comment (yuliya-guseva) [reply java-api-core/src/com/kinvey/java/network/NetworkManager.java]: All Requests classes (Save, Delete etc) are internal classes. It's old logic so I added the new class in the same way as similar classes were added before.

File: java-api-core/src/com/kinvey/java/store/BaseDataStore.java [hunk 4]
```diff
@@ -117,6 +124,7 @@ protected BaseDataStore(AbstractClient client, String collection, Class<T> itemT
         this.storeItemType = itemType;
         if (storeType != StoreType.NETWORK) {
             cache = client.getCacheManager().getCache(collection, itemType, storeType.ttl);
+            queryCache = client.getCacheManager().getCache(QUERY_CACHE_COLLECTION, QueryCacheItem.class, Long.MAX_VALUE);
```
[the remark this answers is not in the document]
Comment (yuliya-guseva) [reply java-api-core/src/com/kinvey/java/store/BaseDataStore.java]: Currently _QueryCache is created in BaseDataStore constructor. 
Sync table is created after the first call to this table via `getCache`. We have 2 possible updates:
1. Create _QueryCache table in the realm after Sync table creating. But in this case, the SDK will just create table but can be a case that this _QueryCache table won't be using.
2. Create _QueryCache table after the first call to this table via `getCache`, in the same way as Sync table is initialized.
I think the second variant is more preferable.

[the remark this answers is not in the document]
Comment (yuliya-guseva) [reply java-api-core/src/com/kinvey/java/store/BaseDataStore.java]: Option #2 was implemented.

**Pull request merged**